platform-sdk/swirlds-config-impl/src/main/java/com/swirlds/config/impl/internal/ConfigurationBuilderImpl.java

```diff
@@ -185,4 +196,19 @@ private <T extends Record> void addConfigDataType(@NonNull final Class<T> type)
         }
         configuration.addConfigDataType(type);
     }
+
+    /**
+     * Sets the value for the config.
+     *
+     * @param propertyName name of the property
+     * @param value        the value
+     * @return the {@link ConfigurationBuilder} instance (for fluent API)
+     */
+    @NonNull
+    public ConfigurationBuilder withValue(@NonNull final String propertyName, @NonNull final Object value) {
+        Objects.requireNonNull(propertyName, "propertyName must not be null");
+        Objects.requireNonNull(value, "value must not be null");
+        properties.put(propertyName, value.toString());
```

Comment:
That is not working. The configuration has Converters to define how a string value is transformed to an object of a specific type. The conversion from object to string is currently not defined but it won't be just `toString()`. That won't work for any `java.time` classes for example. For now the best would be to have `withValue(@NonNull final String propertyName, @NonNull final String value)`. Everything else will make that PR really way more complexe.

Reply:
Removed this method

Reply:
Changed to `withValue(@NonNull final String propertyName, @NonNull final String value)`